src/components/popups/TeamSetting.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect } from "react";
+import "../../styles/popups/TeamSetting.scss";
+import { api, handleError } from "helpers/api";
+import PropTypes from "prop-types";
+import { Button } from "components/ui/Button";
+import { useNavigate, useParams } from "react-router-dom";
+
+const TeamSettings = ({ isOpen, onClose }) => {
+  //const [editMode, setEditMode] = useState(false); EDIT Mode: for extensibility in M4
+  const [teamName, setTeamName] = useState();
+  const [teamDescription, setTeamDescription] = useState();
+  const [teamUUID, setTeamUUID] = useState();
+  const [teamMembers, setTeamMembers] = useState([]);
+  const [inviteURL, setInviteURL] = useState();
+  const [copied, setCopied] = useState("");
+  const { teamId } = useParams();
+  const token = localStorage.getItem("token");
+  const userId = localStorage.getItem("id");
+  const [error, setError] = useState("");
+  const [leaveError, setLeaveError] = useState("");
+  const baseURL = window.location.origin;
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchUserTeam = async () => {
+      try {
+        const response = await api.get(`/api/v1/users/${userId}/teams`, {
+          headers: { Authorization: `${token}` },
+        });
+        const foundTeam = response.data.find(
+          (team) => team.teamId === parseInt(teamId)
+        );
+        setTeamName(foundTeam.name);
+        setTeamDescription(foundTeam.description);
+        setTeamUUID(foundTeam.teamUUID);
+        setInviteURL(`${baseURL}/invitation/${foundTeam.teamUUID}`);
+      } catch (error) {
+        setError("Something went wrong. Please try again");
+        if (error.response.status === 401) {
+          setLeaveError("You are not authorized to do this");
+        }
+      }
+    };
+
+    fetchUserTeam();
+
+    const fetchTeamMembers = async () => {
+      try {
+        const response = await api.get(`/api/v1/teams/${teamId}/users`, {
+          headers: {
+            Authorization: `${token}`,
+          },
+        });
+        setTeamMembers(response.data);
+      } catch (error) {
+        setError("Something went wrong. Please try again");
+        if (error.response.status === 401) {
+          setLeaveError("You are not authorized to do this");
+        }
+      }
```

Comment:
log error (otherwise no chance to see full error

Reply:
done